src/Libraries/Microsoft.Extensions.AI.Abstractions/Tools/HostedMcpServerTool.cs

```diff
@@ -103,7 +103,34 @@ private static string ValidateUrl(Uri serverUrl)
     /// <summary>
     /// Gets or sets the OAuth authorization token that the AI service should use when calling the remote MCP server.
     /// </summary>
-    public string? AuthorizationToken { get; set; }
+    /// <remarks>
+    /// When set, this value is automatically added to the <see cref="Headers"/> dictionary with the key "Authorization" 
+    /// and the value "Bearer {token}". Setting this property will overwrite any existing "Authorization" header in <see cref="Headers"/>.
+    /// Setting this property to <see langword="null"/> will remove the "Authorization" header from <see cref="Headers"/>.
+    /// </remarks>
+    public string? AuthorizationToken
+    {
+        get
+        {
+            if (Headers.TryGetValue("Authorization", out string? value) && value.StartsWith("Bearer ", StringComparison.Ordinal))
+            {
+                return value.Substring("Bearer ".Length);
+            }
+
+            return null;
+        }
+        set
+        {
+            if (value is not null)
+            {
+                Headers["Authorization"] = $"Bearer {value}";
+            }
+            else
+            {
+                _ = Headers.Remove("Authorization");
```

Comment:
"Authorization" is used multiple times. It'd be good to put it into a named private constant.